### PR TITLE
Add companyUri and extension to image upload functionality

### LINKED ISSUE #643: Upload Image to S3 and Store Record with Format in Database
### Describe the enhancement

Create functionality to allow uploading an image to an AWS S3 bucket and store a corresponding record in the database. The database record should include its format (PNG).

### Why should this enhancement be implemented? Add additional informaiton.

Implementing this feature is essential for enabling users or systems to store image files efficiently and reliably in a centralized location (S3), which ensures scalability and availability. 

### Confirmation

- [x] I have searched for similar issues before creating this one.
- [x] I have provided all the necessary information to understand this documentation issue.

### PR DESCRIPTION
**Pull request by shafat730**
## Description

closes #643 
This PR enhances the addCatalogController by generating a companyUri from the request body and including it in the stored image metadata. This provides a URL-friendly identifier that can be used for routing or linking.

I have also added an extension field to the images Mongoose schema, which stores the file type of uploaded images (e.g .png). This enhancement supports better validation, filtering, and content-type handling.
Added a validation schema for company URI.

Changes made in image model

- Added extension as a required String field in the schema.
- Included extension in the .data() method for consistent API responses.
- Updated the newImage static method to include extension in new documents.

## What type of PR is this? (Check all applicable)

- [ ] 🍕 Feature
- [x] 🐛 Bug Fix
- [ ] 📄 Documentation Update
- [x] 👨‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [x] ✅ Test
- [ ] 🛠️ CI/CD

## Screenshots (if applicable)

image

https://github.com/user-attachments/assets/fd9e0316-7c07-4ff3-ac01-56903c600378

## Checklist
- [x] I have performed a self-review of my code  
- [x] I have commented my code, particularly in hard-to-understand areas  
- [x] I have added tests that prove my fix is effective or that my feature works  
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: packages/app/controllers/catalog.js [hunk 1]
```diff
@@ -187,10 +187,11 @@ async function addCatalogController(req, res, next) {
     let { userId } = req.userData;
     const file = req.file;
     const imageSize = file.size;
+    const companyUri = req.body.companyUri;
```
Comment (aps08) [inline packages/app/controllers/catalog.js:191]: where is the validation of companyUri ? What will happen if the user doesn't pass this

**Comment by aps08** [comment]
@shafat730 ?

File: packages/app/__tests__/controllers/catalog/createlogo.js
Comment (aps08) [inline packages/app/__tests__/controllers/catalog/createlogo.js:1]: add a test scenario where body is empty and check if it's working.

File: packages/app/controllers/catalog.js (hunk 1 above)
Comment (aps08) [reply packages/app/controllers/catalog.js:191]: Is this statement going to work when body itself is not present in the request ?

**Review by aps08 — DISMISSED** [review]
